src/vasp/automatedPreprocessing/automated_preprocessing.py

```diff
@@ -431,6 +431,32 @@ def run_pre_processing(input_model, verbose_print, smoothing_method, smoothing_f
                 sys.exit(-1)
         else:
             distance_to_sphere = read_polydata(str(file_name_distance_to_sphere_solid_thickness))
+    elif solid_thickness == "painted":
+        try:
+            import pyvista as pv
+        except ImportError:
+            print("ERROR: Painted solid thickness requires PyVista to be installed.")
+            sys.exit(-1)
+
+        surface_pv = pv.wrap(surface_extended)
+        aneudraw_data = base_path.with_name(base_path.name + "_aneudraw_surface.vtp")
+        assert aneudraw_data.is_file(), "ERROR: Aneudraw file is missing."
+
+        aneudraw = pv.read(aneudraw_data)
+        point_array = pv.point_array(aneudraw, "Thickness")
+        points = pv.PolyData(aneudraw.points)
+        points['Thickness'] = point_array
+
+        # Run the interpolation from the aneudraw data to the surface mesh generated within this script
+        # may need to modify this interpolation radius for each mesh
+        interpolated = surface_pv.interpolate(points, radius=0.4, null_value=solid_thickness_parameters[0])
+        aneudrawInterpolatedFile = base_path.with_name(base_path.name + "_aneudraw_interpolated.vtp")
+        interpolated.save(aneudrawInterpolatedFile)
+        thickness_array = create_vtk_array(interpolated["Thickness"], "Thickness")
+        distance_to_sphere.GetPointData().AddArray(thickness_array)
+        write_polydata(distance_to_sphere, str(file_name_distance_to_sphere_solid_thickness))
+        print("--- Constructed solid thickness from painted surface mesh\n")
+
```

Comment:
It would be nice to avoid requiring PyVista. How about using `vktPointInterpolator` instead? I have tried to implement this here:
``` python
    elif solid_thickness == "painted":
        file_name_aneudraw_data = base_path.with_name(base_path.name + "_aneudraw_surface.vtp")
        assert file_name_aneudraw_data.is_file(), "ERROR: Aneudraw file is missing."
        aneudraw = read_polydata(str(file_name_aneudraw_data))
        thickness_array = aneudraw.GetPointData().GetArray("Thickness")
        
        # Create points for interpolation
        points = vtk.vtkPolyData()
        points.SetPoints(aneudraw.GetPoints())
        points.GetPointData().AddArray(thickness_array)
        
        # Use a Gaussian kernel for interpolation
        gaussian_kernel = vtk.vtkGaussianKernel()
        gaussian_kernel.SetRadius(solid_thickness_parameters[1])
        gaussian_kernel.SetSharpness(solid_thickness_parameters[2])
        
        # Set up the interpolator with a Gaussian kernel
        interpolator = vtk.vtkPointInterpolator()
        interpolator.SetInputData(surface_extended)
        interpolator.SetSourceData(points)
        interpolator.SetNullValue(solid_thickness_parameters[0])
        interpolator.SetKernel(gaussian_kernel)
        interpolator.Update()
        
        interpolated = interpolator.GetOutput()
        aneudraw_interpolated_file = base_path.with_name(base_path.name + "_aneudraw_interpolated.vtp")
        write_polydata(interpolated, str(aneudraw_interpolated_file))
        
        thickness_array_interpolated = interpolated.GetPointData().GetArray("Thickness")
        distance_to_sphere.GetPointData().AddArray(thickness_array_interpolated)
        
        print("--- Constructed solid thickness from painted surface mesh\n")
```
The help text for the `--solid-thickness-parameters` option should also be updated. I have made an attempt at this here:
``` python
parser.add_argument('-stp', '--solid-thickness-parameters',
                    type=float,
                    nargs="+",
                    default=[0.3],
                    help="Parameters for solid thickness [m]. For 'constant' solid thickness, provide a single float. "
                         "For 'variable' solid thickness, provide four floats for the distancetosphere scaling function: "
                         "'offset', 'scale', 'min', and 'max'. For example, --solid-thickness-parameters 0 0.1 0.25 0.3. "
                         "For 'painted' solid thickness, provide three floats: the first value is the null value used "
                         "when no nearby thickness data is available, the second value is the radius of the Gaussian "
                         "kernel for interpolation, and the third value is the sharpness parameter of the Gaussian kernel "
                         "to adjust the falloff rate. For example, --solid-thickness-parameters 0.3 0.4 2.0. "
                         "Note: If --scale-factor is used, 'offset', 'min', and 'max' parameters will be adjusted "
                         "accordingly for 'variable' solid thickness, and the constant value will also be scaled for "
                         "'constant' solid thickness.")
```